plugins/official/stack-snippets/src/common.ts

```diff
@@ -0,0 +1,294 @@
+import { Node as ProsemirrorNode } from "prosemirror-model";
+import { generateRandomId } from "../../../../src/shared/utils";
```

Comment:
I think we should keep references to editor source code to a minimum in plugins and make those references explicit. I would advocate for exposing the module you need in the `src/index.ts` and then I believe you should be even able to reference them here using the package name `import { module } from "@stackoverflow/stacks-editor"`. Basically I am advocating to be much more explicit about the core parts of the editor that are exposed for plugins to use (intentional coupling). If the package reference does not work I would also be ok to use a relative reference to `src/index` for now but the idea is to have a clear layer where we understand what core parts are "public".

This will help us in the long run to evolve the codebase with confidence.